js/core/utils/window.js

```diff
@@ -2,49 +2,40 @@
 
 const domAdapter = require('../dom_adapter');
 
-const hasWindow = typeof window !== 'undefined';
-let windowObject = hasWindow && window;
+const windowObject = window ?? { window: windowObject };
 
-if(!windowObject) {
-    windowObject = {};
-    windowObject.window = windowObject;
-}
+const hasWindow = () => typeof window !== 'undefined';
 
-module.exports = {
-    hasWindow: function() {
-        return hasWindow;
-    },
-
-    getWindow: function() {
-        return windowObject;
-    },
-
-    hasProperty: function(prop) {
-        return this.hasWindow() && prop in windowObject;
-    },
-
-    defaultScreenFactorFunc: function(width) {
-        if(width < 768) {
-            return 'xs';
-        } else if(width < 992) {
-            return 'sm';
-        } else if(width < 1200) {
-            return 'md';
-        } else {
-            return 'lg';
-        }
-    },
-
-    getCurrentScreenFactor: function(screenFactorCallback) {
-        const screenFactorFunc = screenFactorCallback || this.defaultScreenFactorFunc;
-        const windowWidth = domAdapter.getDocumentElement()['clientWidth'];
-
-        return screenFactorFunc(windowWidth);
-    },
-
-    getNavigator: function() {
-        return this.hasWindow() ? windowObject.navigator : {
-            userAgent: ''
-        };
+const getWindow = () => windowObject;
+
+const hasProperty = (prop) => hasWindow() && prop in windowObject;
+
+const defaultScreenFactorFunc = (width) => {
+    if(width < 768) {
+        return 'xs';
+    } else if(width < 992) {
+        return 'sm';
+    } else if(width < 1200) {
+        return 'md';
+    } else {
+        return 'lg';
     }
 };
+
+const getCurrentScreenFactor = (screenFactorCallback) => {
+    const screenFactorFunc = screenFactorCallback || defaultScreenFactorFunc;
+    const windowWidth = domAdapter.getDocumentElement()['clientWidth'];
+
+    return screenFactorFunc(windowWidth);
+};
+
+const getNavigator = () => hasWindow() ? windowObject.navigator : { userAgent: '' };
+
+module.exports = {
+    hasWindow,
+    getWindow,
```

Comment:
Why aren't we using `export const ...` ?

Reply:
In some places we use
```js
const windowUtils = require('../core/utils/window');
```
This is incompatible with
```js
export const hasWindow = ...;
export const getWindow = ...;
```
or with
```js
export {
    hasWindow,
    getWindow,
    ...
}
```

If we want to use it, we should get rid of all `windowUtils = require`. Should we do that in this PR?

Reply:
I think we want to do it.

> Should we do that in this PR?

Yes, we can.

Reply:
We have too many places, where we use `require`. It's quite difficult to change them all in same time (many dependencies between components).